account_move_base_import/models/account_journal.py

```diff
@@ -0,0 +1,396 @@
+        """
+        filename = self._context.get("file_name", None)
+        attachment_obj = self.env["ir.attachment"]
+        if filename:
+            (filename, __) = os.path.splitext(filename)
+        parser = new_move_parser(self, ftype=ftype, move_ref=filename)
+        res = self.env["account.move"]
+        for result_row_list in parser.parse(file_stream):
+            move = self._move_import(
+                parser,
+                file_stream,
+                result_row_list=result_row_list,
+                ftype=ftype,
+            )
+            res |= move
+        if res:
+            attachment_vals = self._get_attachment_data(res, file_stream, ftype)
+            if attachment_vals:
+                attachment_obj.create(attachment_vals)
+        return res
+
+    def _move_import(self, parser, file_stream, result_row_list=None, ftype="csv"):
+        """Create a bank statement with the given profile and parser. It will
+        fulfill the bank statement with the values of the file provided, but
+        will not complete data (like finding the partner, or the right
+        account). This will be done in a second step with the completion rules.
+
+        :param prof : The profile used to import the file
+        :param parser: the parser
+        :param filebuffer file_stream: binary of the provided file
+        :param char: ftype represent the file extension (csv by default)
+        :return: ID of the created account.bank.statement
+        """
+        move_obj = self.env["account.move"]
+        move_line_obj = self.env["account.move.line"]
+        attachment_obj = self.env["ir.attachment"]
+        if result_row_list is None:
+            result_row_list = parser.result_row_list
+        # Check all key are present in account.bank.statement.line!!
+        if not result_row_list:
+            raise UserError(_("Nothing to import: " "The file is empty"))
+        parsed_cols = list(parser.get_move_line_vals(result_row_list[0]).keys())
+        for col in parsed_cols:
+            if col not in move_line_obj._fields:
+                raise UserError(
+                    _(
+                        "Missing column! Column %s you try to import is not "
+                        "present in the move line!"
+                    )
+                    % col
+                )
+        move_vals = self.prepare_move_vals(result_row_list, parser)
+        move = move_obj.create(move_vals)
+        try:
+            # Record every line in the bank statement
+            move_store = []
+            for line in result_row_list:
+                parser_vals = parser.get_move_line_vals(line)
+                values = self.prepare_move_line_vals(parser_vals, move)
+                move_store.append(values)
+            # move_line_obj.with_context(check_move_validity=False).create(move_store)
+            move_store += self._get_extra_move_line_vals_list(parser, move)
```

Comment:
i would just delete the line instead of commenting, but not particularly a big deal